tutorials/scripting/gdextension/gdextension_c_example.rst

```diff
@@ -0,0 +1,2154 @@
+    target_path = "demo/bin/"
+    target_name = "libgdexample"
+
+    # Set the compiler and flags.
+    env.Append(CPPPATH=["src"])  # Add the src folder to the include path.
+    env.Append(CFLAGS=["-O0", "-g"])  # Make it a debug build.
+
+    # Use Clang on macOS.
+    if sys.platform == "darwin":
+        env["CC"] = "clang"
+
+    # Add all C files in "src" folder as sources.
+    sources = env.Glob("src/*.c")
+
+    # Create a shared library.
+    library = env.SharedLibrary(
+        target=path.join(target_path, target_name),
+        source=sources,
+    )
+
+    # Set the library as the default target.
+    env.Default(library)
+
+This will include all C files in the ``src`` folder, so we don't need to change
+this file when adding new source files.
+
+Initializing the extension
+--------------------------
+
+The first bit of code will be responsible to initialize the extension. This is
+what makes Godot aware of what our GDExtension provides, such as classes and
+plugins.
+
+Create the file ``init.h`` in the ``src`` folder, with the following contents:
+
+.. code-block:: c
+
+    #ifndef INIT_H
+    #define INIT_H
+
+    #include "defs.h"
+
+    #include "gdextension_interface.h"
+
+    void initialize_gdexample_module(void *p_userdata, GDExtensionInitializationLevel p_level);
+    void deinitialize_gdexample_module(void *p_userdata, GDExtensionInitializationLevel p_level);
+    GDExtensionBool GDE_EXPORT gdexample_library_init(GDExtensionInterfaceGetProcAddress p_get_proc_address, GDExtensionClassLibraryPtr p_library, GDExtensionInitialization *r_initialization);
+
+    #endif // INIT_H
+
+The functions declared here have the signature expected by the GDExtension API.
+
+Note the inclusion of the ``defs.h`` file. This is one of our helpers to
+simplify writing the extension code. For now it will only contain the definition
+of ``GDE_EXPORT``, a macro that makes the function public in the shared library
+so Godot can properly call it. This macro helps abstracting what each compiler
+expects.
+
+Create the ``defs.h`` file in the ``src`` folder with the following contents:
+
+.. code-block:: c
+
+    #ifndef DEFS_H
+    #define DEFS_H
+
+    #include <stdbool.h>
+    #include <stddef.h>
+    #include <stdint.h>
+
+    #if !defined(GDE_EXPORT)
+    #if defined(_WIN32)
+    #define GDE_EXPORT __declspec(dllexport)
+    #elif defined(__GNUC__)
+    #define GDE_EXPORT __attribute__((visibility("default")))
+    #else
+    #define GDE_EXPORT
+    #endif
+    #endif // ! GDE_EXPORT
+
+    #endif // DEFS_H
+
+We also include some standard headers to make things easier. Now we only have to
+include ``defs.h`` and those will come as a bonus.
+
+Now, let's implement the functions we just declared. Create a file called
+``init.c`` in the ``src`` folder and add this code:
+
+.. code-block:: c
+
+    #include "init.h"
+
+    void initialize_gdexample_module(void *p_userdata, GDExtensionInitializationLevel p_level)
+    {
```

Comment:
I'd suggest following the code style for this code as well, with brackets on the declaration